kolibri/core/assets/src/utils/errorReportUtils.js

```diff
@@ -6,13 +6,35 @@ class ErrorReport {
   getErrorReport() {
     throw new Error('getErrorReport() method must be implemented.');
   }
+  getDeviceInfo() {
+    return {
+      type: /Mobi|Android/i.test(navigator.userAgent) ? 'Mobile' : 'Desktop',
+      platform: navigator.platform,
+      screen: {
+        width: window.innerWidth,
+        height: window.innerHeight,
+      },
+    };
+  }
+  getBrowserInfo() {
+    return navigator.userAgent;
+  }
 }
 
 export class VueErrorReport extends ErrorReport {
+  constructor(e, vm) {
+    super(e);
+    this.vm = vm;
+  }
   getErrorReport() {
     return {
       error_message: this.e.message,
       traceback: this.e.stack,
+      context: {
+        component: this.vm.$options.name || this.vm.$options._componentTag || 'Unknown Component',
+        browser: this.getBrowserInfo(),
+        device: this.getDeviceInfo(),
+      },
```

Comment:
`context` can be movet to ErrorReport too. We can then do something like the below;

```js
getErrorReport() {
    const context = this.getContext();
    return {
      ...
      context: {
        component: this.vm.$options.name || this.vm.$options._componentTag || 'Unknown Component',
        ...context,
      },
    };
  }
```

Reply:
Could even not even need to define `getErrorReport` in this class, instead can just define it in the parent class once, and then here we do:
```
getContext() {
   const context = super.getContext();
   return {
      ...context,
      component <etc>...
   };
}
```